packages/jest-mock/src/__tests__/index.test.ts

```diff
@@ -12,6 +12,14 @@ import * as util from 'util';
 import {type Context, createContext, runInContext, runInNewContext} from 'vm';
 import {ModuleMocker, fn, mocked, spyOn} from '../';
 
+if (!Symbol.dispose) {
+  Object.defineProperty(Symbol, 'dispose', {
+    get() {
+      return Symbol.for('nodejs.dispose');
```

Comment:
This is a bit weird, and the result of #14888 - usually, it should be possible to polyfill `Symbol.dispose` in any possible way, e.g. by doing `Symbol.dispose ??= Symbol.for('dispose')`, but because of #14888 it needs to explicitly be polyfilled with `Symbol.for('nodejs.dispose')`.  

I'm not sure if that change over there was a good idea, especially because that polyfill doesn't seem to be available in all scopes (that's why we need to polyfill here in the first place).

Reply:
I think that also explains the failing node16 tests.

Reply:
that's what the symbol is in a nodejs app, tho. so why would it be an issue?

But if we instead move the test to an e2e test as mentioned elsewhere, then we could just skip that tests on node versions which don't support it

Reply:
It was a bit irritating to me first because jest picks up `Symbol.dispose` as `Symbol.for('nodejs.dispose')`, while in the test itself was `undefined`, so it needed to be polyfilled to the same value (while in a normal application, you can polyfill it as pretty much anything and don't need to match the "outside runtime").

But I guess it makes sense that `jest` itself might see something different than the JestEnvironment itself.